SMLHelper/Options/ChoiceModOption.cs

```diff
@@ -155,17 +157,50 @@ public class ModChoiceOption : ModOption
         /// </summary>
         public int Index { get; }
 
+        internal override void AddToPanel(uGUI_TabbedControlsPanel panel, int tabIndex)
+        {
+            var choice = panel.AddChoiceOption(tabIndex, Label, Options, Index,
+                new UnityAction<int>((int index) => parentOptions.OnChoiceChange(Id, index, Options[index])));
+
+            OptionGameObject = choice.transform.parent.transform.parent.gameObject; // :(
+
+            base.AddToPanel(panel, tabIndex);
+        }
+
         /// <summary>
         /// Instantiates a new <see cref="ModChoiceOption"/> for handling an option that can select one item from a list of values.
         /// </summary>
         /// <param name="id">The internal ID of this option.</param>
         /// <param name="label">The display text to show on the in-game menus.</param>
         /// <param name="options">The collection of available values.</param>
         /// <param name="index">The starting value.</param>
-        internal ModChoiceOption(string id, string label, string[] options, int index) : base(ModOptionType.Choice, label, id)
+        internal ModChoiceOption(string id, string label, string[] options, int index) : base(label, id)
         {
             this.Options = options;
             this.Index = index;
         }
+
+        private class ChoiceOptionAdjust: ModOptionAdjust
+        {
+            private const float spacing = 10f;
+
+            public IEnumerator Start()
+            {
+                SetCaptionGameObject("Choice/Caption");
+                yield return null; // skip one frame
+
+                RectTransform rect = gameObject.transform.Find("Choice/Background") as RectTransform;
+
+                float widthAll = gameObject.GetComponent<RectTransform>().rect.width;
+                float widthChoice = rect.rect.width;
+                float widthText = CaptionWidth + spacing;
+
+                if (widthText + widthChoice > widthAll)
+                    rect.sizeDelta = SetVec2x(rect.sizeDelta, widthAll - widthText - widthChoice);
```

Comment:
Is this certain enough to never need null checking?
```
RectTransform rect = gameObject.transform.Find("Choice/Background") as RectTransform;
```

Reply:
Yea, we talked about it: https://github.com/SubnauticaModding/SMLHelper/pull/99#issuecomment-548503982